### PR TITLE
[tune] Change tune resource request syntax to be less confusing

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

This renames resources => trial_resources, and cpu / driver_cpu_limit => cpu / extra_cpu.

## Related issue number

https://github.com/ray-project/ray/issues/1759

### CODE REVIEW

File: python/ray/tune/config_parser.py [hunk 1]
```diff
@@ -56,7 +66,12 @@ def make_parser(**kwargs):
...
     parser.add_argument(
-        "--resources", default='{"cpu": 1}', type=json_to_resources,
+        "--resources", help="Deprecated, use --trial-resources.",
+        type=lambda v: _tune_error(
+            "The `resources` argument is no longer supported. "
+            "Use `trial_resources` or --trial-resources instead."))
+    parser.add_argument(
+        "--trial-resources", default='{"cpu": 1}', type=json_to_resources,
         help="Machine resources to allocate per trial, e.g. "
         "'{\"cpu\": 64, \"gpu\": 8}'. Note that GPUs will not be assigned "
```
Comment (richardliaw) [inline python/ray/tune/config_parser.py]: I think this needs to be fixed

File: python/ray/tune/experiment.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ class Experiment(object):
             empty dict.
         config (dict): Algorithm-specific configuration
             (e.g. env, hyperparams). Defaults to empty dict.
-        resources (dict): Machine resources to allocate per trial,
+        trial_resources (dict): Machine resources to allocate per trial,
             e.g. ``{"cpu": 64, "gpu": 8}``. Note that GPUs will not be
```
Comment (richardliaw) [inline python/ray/tune/experiment.py]: same with this

File: python/ray/tune/config_parser.py (hunk 1 above)
Comment (richardliaw) [inline python/ray/tune/config_parser.py]: I think this needs to be fixed

File: python/ray/rllib/train.py [hunk 3]
```diff
@@ -62,7 +62,7 @@
                 "run": args.run,
                 "checkpoint_freq": args.checkpoint_freq,
                 "local_dir": args.local_dir,
-                "resources": resources_to_json(args.resources),
+                "trial_resources": resources_to_json(args.resources),
```
Comment (richardliaw) [inline python/ray/rllib/train.py]: args.trial_resources?

File: examples/carla/train_a3c.py [hunk 4]
```diff
@@ -32,7 +32,7 @@
     "carla-a3c": {
         "run": "A3C",
         "env": "carla_env",
-        "resources": {"cpu": 5, "gpu": 2, "driver_gpu_limit": 0},
+        "trial_resources": {"cpu": 5, "gpu": 2, "driver_gpu_limit": 0},
```
Comment (richardliaw) [inline examples/carla/train_a3c.py]: driver_gpu_limit deprecated

**Review by richardliaw — CHANGES_REQUESTED** [review]

**Review by ericl — COMMENTED** [review]
Updated

**Review by richardliaw — APPROVED** [review]

**Pull request merged**